bbconf/config_parser/bedbaseconfig.py

```diff
@@ -0,0 +1,338 @@
+        :return: None
+        :raises: raise_missing_key (if config key is missing)
+        """
+        _config = yacman.YAMLConfigManager(filepath=config_path).exp
+
+        config_dict = {}
+        for field_name, annotation in ConfigFile.model_fields.items():
+            try:
+                config_dict[field_name] = annotation.annotation(
+                    **_config.get(field_name)
+                )
+            except TypeError:
+                # TODO: this should be more specific
+                config_dict[field_name] = annotation.annotation()
+
+        return ConfigFile(**config_dict)
+
+    @property
+    def config(self) -> ConfigFile:
+        """
+        Get configuration
+
+        :return: configuration object
+        """
+        return self._config
+
+    @property
+    def db_engine(self) -> BaseEngine:
+        """
+        Get database engine
+
+        :return: database engine
+        """
+        return self._db_engine
+
+    @property
+    def t2bsi(self) -> Union[text2bednn.Text2BEDSearchInterface, None]:
+        """
+        Get text2bednn object
+
+        :return: text2bednn object
+        """
+        return self._t2bsi
+
+    @property
+    def r2v(self) -> Region2VecExModel:
+        """
+        Get region2vec object
+
+        :return: region2vec object
+        """
+        return self._r2v
+
+    @property
+    def qdrant_engine(self) -> QdrantBackend:
+        """
+        Get qdrant engine
+
+        :return: qdrant engine
+        """
+        return self._qdrant_engine
+
+    @property
+    def phc(self) -> PEPHubClient:
+        """
+        Get PEPHub client
+
+        :return: PEPHub client
+        """
+        return self._phc
+
+    @property
+    def boto3_client(self) -> boto3.client:
+        """
+        Get boto3 client
+
+        :return: boto3 client
+        """
+        return self._boto3_client
+
+    def _init_db_engine(self) -> BaseEngine:
+        return BaseEngine(
+            host=self._config.database.host,
+            port=self._config.database.port,
+            database=self._config.database.database,
+            user=self._config.database.user,
+            password=self._config.database.password,
+            drivername="postgresql+psycopg",
+        )
+
+    def _init_qdrant_backend(self) -> QdrantBackend:
+        """
+        Create qdrant client object using credentials provided in config file
+
+        :return: QdrantClient
+        """
+        try:
+            return QdrantBackend(
+                collection=self._config.qdrant.collection,
+                qdrant_host=self._config.qdrant.host,
+                qdrant_port=self._config.qdrant.port,
+                qdrant_api_key=self._config.qdrant.api_key,
+            )
+        except qdrant_client.http.exceptions.ResponseHandlingException as err:
+            _LOGGER.error(f"error in Connection to qdrant! skipping... Error: {err}")
+            warnings.warn(
```

Comment:
Why the warning and the error?

Reply:
Warning is more strong then just logger error, and in tests or code you will see that something is wrong.